text/0000-opentelemetry.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: default
+title: OpenTelemetry Integration
+nav_order: 3
+---
+
+- Feature Name: OpenTelemetry Integration
+- Start Date: 2021-01-12
+- RFC PR: (leave this empty)
+- Fabric Component: core, chaincode, sdks
+- Fabric Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This request for comments proposes integrating Hyperledger Fabric, its SDKs, core and chaincode components 
+with the OpenTelemetry project. It advocates for relying on OpenTelemetry to record and send metrics.
+It introduces the concept of tracing, reporting the execution of chaincode, core components and SDKs to help correlate activities with the chain.
+
+# Motivation
+[motivation]: #motivation
+
+The OpenTelemetry project is a CNCF project that aims to standardize observability, especially in a cloud-native environment.
+OpenTelemetry is working to create a standard around representing metrics, traces and logs.
+We aim to bring full observability of Hyperledger Fabric to allow tracing of transactions all the way from the client
+to each of the chaincode invocations. We also want to capture chaincode interactions.
```

Comment:
>  We also want to capture chaincode interactions.

Why is this information helpful?

> We want to be able to trace and correlate actions from chaincode events.

A chaincode event is simply an output of its execution. It is recorded on the blockchain. You can't trace it as it doesn't go to any node.

Reply:
This information is helpful as it helps understand the path of execution of code - the equivalent of logging entering methods and exiting methods, but with a simple, lightweight object that can track metadata and tags for the purpose of powering operations.

Chaincode events are typically triggers for oracles that then spawn code execution. Using the trace information associated with the event allows to correlate the execution of arbitrary code back to the event that triggered it and the transaction that originated the event.